File: frontend/src/types/stress_test.ts
export interface ThrottleReading {
  elapsed_seconds: number;
  throttled_hex: string;
  undervoltage_now: boolean;
  freq_capped_now: boolean;
  throttled_now: boolean;
  undervoltage_occurred: boolean;
  throttled_occurred: boolean;
  temp_celsius: number | null;
  volts: number | null;
}

export type StressVerdict =
  | "clean"
  | "undervoltage_under_load"
  | "throttled_under_load"
  | "crashed_or_unreachable"
  | "unknown";

export interface NodeStressResult {
  node_name: string;
  ip_address: string;
  reachable: boolean;
  error: string | null;
  ncores: number | null;
  baseline: ThrottleReading | null;
  samples: ThrottleReading[];
  final: ThrottleReading | null;
  max_temp_celsius: number | null;
  min_volts: number | null;
  undervoltage_detected: boolean;
  throttled_detected: boolean;
  verdict: StressVerdict;
}

export interface StressTestReport {
  started_at: string;
  duration_seconds: number;
  nodes: NodeStressResult[];
  any_undervoltage: boolean;
  any_node_unreachable: boolean;
  summary: string;
}

File: backend/app/schemas/stress_test.py
from datetime import datetime

from pydantic import BaseModel, Field


class StressTestRequest(BaseModel):
    duration_seconds: int = Field(default=30, ge=10, le=120)


class ThrottleReading(BaseModel):
    elapsed_seconds: float
    throttled_hex: str
    undervoltage_now: bool
    freq_capped_now: bool
    throttled_now: bool
    undervoltage_occurred: bool
    throttled_occurred: bool
    temp_celsius: float | None
    volts: float | None


class NodeStressResult(BaseModel):
    node_name: str
    ip_address: str
    reachable: bool
    error: str | None = None
    ncores: int | None = None
    baseline: ThrottleReading | None = None
    samples: list[ThrottleReading] = []
    final: ThrottleReading | None = None
    max_temp_celsius: float | None = None
    min_volts: float | None = None
    undervoltage_detected: bool = False
    throttled_detected: bool = False
    verdict: str = "unknown"


class StressTestReport(BaseModel):
    started_at: datetime
    duration_seconds: int
    nodes: list[NodeStressResult]
    any_undervoltage: bool
    any_node_unreachable: bool
    summary: str
